msal/application.py

```diff
@@ -511,7 +511,7 @@ def acquire_token_by_device_flow(self, flow, **kwargs):
                 **kwargs)
 
     def acquire_token_by_username_password(
-            self, username, password, scopes=None, **kwargs):
+            self, username, password, scopes=[], **kwargs):
```

Comment:
I understand your point. However this fix is [not ideal](https://docs.quantifiedcode.com/python-anti-patterns/correctness/mutable_default_value_as_argument.html). I would suggest to rollback this change in current PR, so that it would only fix that `verify` (and `proxies` and `timeout` issue) and get merged soon, and we will come up with a separated PR to address that scope thing.

Reply:
Yes I meant further down the stack inside MSAL. My application is supplying a scope I only noticed the crash when I was experimenting. As a rule I think libraries should not crash if valid (according to the interface) arguments are supplied.

But agreed I'll remove that fix and put in timeout and proxies as well in addition to verify.

Reply:
The "default value" topic is addressed in PR #64. Closing this conversation.